File: lektion 2/app/src/main/java/com/example/lektion2/MainActivity.java
package com.example.lektion2;

import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;



public class MainActivity extends AppCompatActivity {
    @Override
    protected void onStart() {
        super.onStart();
        Log.i("Aleksander", "onStart: ");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.i("Aleksander", "onStop: ");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.i("Aleksander", "onDestroy: ");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d("Aleksander", "onPause: ");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i("Aleksander", "onResume: ");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.i("Aleksander", "onRestart: ");
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        TextView tv1 = findViewById(R.id.textView2);
        ImageView iv = findViewById(R.id.imageViewDroid);

       // tv1.setText(R.string.lang);
        tv1.setText(getResources().getStringArray(R.array.system)[1]);
        tv1.setTextColor(getResources().getColor(R.color.purple));
        //samma sak som ovanför
        tv1.setTextColor(ContextCompat.getColor(MainActivity.this,R.color.purple));

        iv.setAlpha(0.2f);
        iv.setRotationY(170);
        iv.setImageResource(R.drawable.img_1033);


        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }
}

File: lektion 4.2/app/src/main/java/com/example/lektion42/MainActivity.kt
package com.example.lektion42

import android.content.ContentValues.TAG
import android.os.Bundle
import android.util.Log
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.google.firebase.Firebase
import com.google.firebase.firestore.firestore

class MainActivity : AppCompatActivity() {

    var db = Firebase.firestore

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        write()
        read()
        delete()

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }

    fun write() {
        val user = hashMapOf(
            "first" to "Ada",
            "last" to "Lovelace",
            "born" to 1815,
        )


        db.collection("users")
            .add(user)
            .addOnSuccessListener { documentReference ->
                Log.d("Aleksander", "DocumentSnapshot added with ID: ${documentReference.id}")
            }
            .addOnFailureListener { e ->
                Log.w("Aleksander", "Error adding document", e)
            }
    }

    fun read(){
        db.collection("users")
            .get()
            .addOnSuccessListener { result ->
                for (document in result) {
                    Log.d(TAG, "${document.id} => ${document.data}")
                }
            }
            .addOnFailureListener { exception ->
                Log.w(TAG, "Error getting documents.", exception)
            }
    }

    fun delete(){
        db.collection("users").document("ipCuJBKmVGkt2fNWZ67M")
            .delete()
            .addOnSuccessListener { Log.d(TAG, "DocumentSnapshot successfully deleted!") }
            .addOnFailureListener { e -> Log.w(TAG, "Error deleting document", e) }
    }
}